MongoDB/todoApp.js

```diff
@@ -22,88 +22,175 @@ MongoClient.connect(url, function(err, database){
+    });
+    }
 });
+
 app.get('/work/:id', function(req, res){
-    var query = { ID: parseInt(req.params.id)};
-    db.collection('work').find(query).toArray(function(err, docs){
-        res.send(docs);
-      });
+    if(isNaN(parseInt(req.params.id)) == true){
+      res.send('Wrong Input!!! (Maybe it is not number)', 400);
+    }
+    else{
+    var myquery = { ID: parseInt(req.params.id)};
+    db.collection('work').findOne(myquery,function(err, docs){
+      if(docs == null){
+        res.send('Not found :-(', 404);
+      } else {
+        res.send(docs);}
+    });
+    }
 });
 
 //delete
-app.delete('/task/:id', function(req, res) {
+app.delete('/task/:id', function(req, res){
+  if(isNaN(parseInt(req.params.id)) == true){
+    res.send('Wrong Input!!! (Maybe it is not number)', 400);
+  }
+  else{
   var myquery = { ID: parseInt(req.params.id)};
-  db.collection('task').remove(myquery, function(err, docs) {
-    db.collection('task').find(myquery).toArray(function(err, docs){
-        res.send(docs);
+  db.collection('task').remove(myquery, function(err, docs){
+    db.collection('task').findOne(myquery,function(err, docs){
+        res.send(docs,204);
     });
   });
+  }
 });
-app.delete('/work/:id', function(req, res) {
+
+app.delete('/work/:id', function(req, res){
+  if(isNaN(parseInt(req.params.id)) == true){
+    res.send('Wrong Input!!! (Maybe it is not number)', 400);
+  }
+  else{
   var myquery = { ID: parseInt(req.params.id)};
-  db.collection('work').remove(myquery, function(err, docs) {
-    db.collection('work').find(myquery).toArray(function(err, docs){
-        res.send(docs);
+  db.collection('work').remove(myquery, function(err, docs){
+    db.collection('work').findOne(myquery,function(err, docs){
+        res.send(docs,204);
     });
   });
+  }
 });
 
 //update
-app.patch('/task/:id/:description', function(req, res){
+app.patch('/task/:id', function(req, res){
+  var json = req.body;
+  if(isNaN(parseInt(req.params.id)) == true){
+    res.send('Wrong Input!!! (Maybe it is not number)', 400);
+  } else {
   var myquery = { ID: parseInt(req.params.id)};
-  var newquery = {$set: {Description: req.params.description}}
-  db.collection('task').findOneAndUpdate(myquery,newquery,{upsert: true},function(err,docs){
-      db.collection('task').find(myquery).toArray(function(err, docs){
-          res.send(docs);
+  var newquery = {$set: {Description: json.Description,
+                        Completed: json.Completed,
+                        Duration: json.Duration}}
+    if(typeof json.Description !== 'string'||typeof json.Completed !== 'boolean'||typeof json.Duration !== 'string'){
+      res.send('Something wrong???', 400);}
+    else{
+      db.collection('task').findOne(myquery,function(err, docs){
+        if(docs == null){
+          res.send('Not found :-(', 404);
+        }
+        else{
+          db.collection('task').findOneAndUpdate(myquery,newquery,{upsert: true},function(err,docs){
```

Comment:
upsert will create a new document if there is no document that matches query (myquery)